### PR TITLE
Ask for username when the given access token is a reference token

### PR DESCRIPTION
**Pull request by sverdlov93**
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

1. On interactive config add - ask for username when reference token is provided (optional) to support build tool commands
2. On CI config add - add a warning log that without a username some build tools wouldn't work
3. On build tool commands (jf pip install for example) - add a warning log if a reference token without a username was provided.

depends on https://github.com/jfrog/jfrog-client-go/pull/674

### CODE REVIEW

File: common/commands/config.go [hunk 1]
```diff
@@ -154,17 +154,29 @@ func (cc *ConfigCommand) config() error {
...
-		} else {
-			cc.details.Url = clientutils.AddTrailingSlashIfNeeded(cc.details.Url)
-			// Derive JFrog services URLs from platform URL
-			coreutils.SetIfEmpty(&cc.details.ArtifactoryUrl, cc.details.Url+"artifactory/")
-			coreutils.SetIfEmpty(&cc.details.DistributionUrl, cc.details.Url+"distribution/")
-			coreutils.SetIfEmpty(&cc.details.XrayUrl, cc.details.Url+"xray/")
-			coreutils.SetIfEmpty(&cc.details.MissionControlUrl, cc.details.Url+"mc/")
-			coreutils.SetIfEmpty(&cc.details.PipelinesUrl, cc.details.Url+"pipelines/")
+	} else {
+		// non-interactive config
```
Comment (eyalbe4) [inline common/commands/config.go:158]: non --> Non

File: common/commands/config.go [hunk 2]
```diff
@@ -154,17 +154,29 @@ func (cc *ConfigCommand) config() error {
...
+				// Derive JFrog services URLs from platform URL
+				coreutils.SetIfEmpty(&cc.details.ArtifactoryUrl, cc.details.Url+"artifactory/")
+				coreutils.SetIfEmpty(&cc.details.DistributionUrl, cc.details.Url+"distribution/")
+				coreutils.SetIfEmpty(&cc.details.XrayUrl, cc.details.Url+"xray/")
+				coreutils.SetIfEmpty(&cc.details.MissionControlUrl, cc.details.Url+"mc/")
+				coreutils.SetIfEmpty(&cc.details.PipelinesUrl, cc.details.Url+"pipelines/")
+			}
+		}
+
+		// Some build tools support basic authentication only,
```
Comment (eyalbe4) [inline common/commands/config.go:174]: ...only, --> ...only.

File: common/commands/config.go [hunk 3]
```diff
@@ -154,17 +154,29 @@ func (cc *ConfigCommand) config() error {
...
+		// to support them we try to extract the username from the access token,
```
Comment (eyalbe4) [inline common/commands/config.go:175]: to --> To

File: common/commands/config.go [hunk 4]
```diff
@@ -154,17 +154,29 @@ func (cc *ConfigCommand) config() error {
...
+		// or request the username interactively
```
Comment (eyalbe4) [inline common/commands/config.go:176]: Since we're inbside the "non-interactive config" flow, do you think we should mention the interactive flow here?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Review by yahavi — CHANGES_REQUESTED** [review]
Let's add a unit test in config_test.go

File: common/commands/config_test.go [hunk 5]
```diff
@@ -49,7 +49,14 @@ func TestArtifactorySshKey(t *testing.T) {
 
 func TestAccessToken(t *testing.T) {
 	inputDetails := tests.CreateTestServerDetails()
-	inputDetails.AccessToken = "accessToken"
```
Comment (yahavi) [inline common/commands/config_test.go:52]: Notice that you deleted the access token's content. Is it intended?
Same question for the new TestAccessTokenWithUsername test.

**Pull request merged**